starknet_py/net/account/account.py

```diff
@@ -344,6 +392,54 @@ async def sign_invoke_v1(
         signature = self.signer.sign_transaction(execute_tx)
         return _add_signature_to_transaction(execute_tx, signature)
 
+    async def sign_outside_execution_call(
+        self,
+        calls: Calls,
+        execution_time_bounds: ExecutionTimeBounds,
+        *,
+        caller: AddressRepresentation = ANY_CALLER,
+        nonce: Optional[int] = None,
+        version: Optional[SNIP9InterfaceVersion] = None,
```

Comment:
1. What's the reason that these need to be passed as keyword arguments?
2. nit: Maybe let's rename `version` to `interface_version`, so the name is a bit more specific?

Reply:
1. To allow for customisation and testing
2. :+1: